src/test/java/com/microsoft/sqlserver/jdbc/AlwaysEncrypted/MultiUserAKVTest.java

```diff
@@ -0,0 +1,643 @@
+
+        boolean result = (boolean)method.invoke(fieldValue, encryptedCEKHexString);
+        
+        return result;
+    }
+
+    private byte[] signColumnMasterKeyMetadata(SQLServerColumnEncryptionAzureKeyVaultProvider provider, String masterKeyPath, boolean allowEnclaveComputations) throws Exception {
+
+        MessageDigest md = MessageDigest.getInstance("SHA-256");
+        md.update(Constants.AZURE_KEY_VAULT_NAME.toLowerCase().getBytes(java.nio.charset.StandardCharsets.UTF_16LE));
+        md.update(masterKeyPath.toLowerCase().getBytes(java.nio.charset.StandardCharsets.UTF_16LE));
+        
+        if (allowEnclaveComputations) {
+            md.update("true".getBytes(java.nio.charset.StandardCharsets.UTF_16LE));
+        } else {
+            md.update("false".getBytes(java.nio.charset.StandardCharsets.UTF_16LE));
+        }
+
+        byte[] dataToSign = md.digest();
+        if (null == dataToSign) {
+            fail("data to sign is null or empty.");
+        }
+
+        assertTrue(dataToSign.length > 0);
+
+        Method method = provider.getClass().getDeclaredMethod("AzureKeyVaultSignHashedData", byte[].class, String.class);
+        method.setAccessible(true);
+
+        byte[] signature = (byte[]) method.invoke(provider, dataToSign, masterKeyPath);
+
+        if (null == signature || 0 == signature.length) {
+            fail("Signature of column master key metadata is null or empty.");
+        }
+
+        return signature;
+    }
+    
+    private int getCacheSize(String methodName, SQLServerColumnEncryptionAzureKeyVaultProvider provider) throws Exception {
+        Method method = provider.getClass().getDeclaredMethod(methodName);
+        method.setAccessible(true);
+        
+        int count = (int)method.invoke(provider);
+
+        return count;     
+    }
+
+    private SQLServerColumnEncryptionAzureKeyVaultProvider createAKVProvider() throws Exception {
+        
+        SQLServerColumnEncryptionAzureKeyVaultProvider azureKeyVaultProvider = null;
+
+        if (null != applicationClientID && null != applicationKey) {
+            File file = null;
+            try {
+                file = new File(Constants.MSSQL_JDBC_PROPERTIES);
+                try (OutputStream os = new FileOutputStream(file);) {
+                    Properties props = new Properties();
+                    // Append to the list of hardcoded endpoints
+                    props.setProperty(Constants.AKV_TRUSTED_ENDPOINTS_KEYWORD, ";vault.azure.net");
+                    props.store(os, "");
+                }
+                azureKeyVaultProvider = new SQLServerColumnEncryptionAzureKeyVaultProvider(applicationClientID, applicationKey);
+                
+            } finally {
+                if (null != file) {
+                    file.delete();
+                }
+            }
+        }
+
+        return azureKeyVaultProvider;
+    }
+
+    private static String byteToHexDisplayString(byte[] b) {

```

Comment:
can we use bytesToHexString in TestUtil? if not, this should go there

Reply:
The return values are slightly different. I have moved it to TestUtils.java